app/models/job_execution.rb

```diff
@@ -179,26 +201,27 @@ def lock_project(&block)
 
   class << self
     def find_by_id(id)
-      registry[id.to_i]
+      registry.find(id)
     end
 
-    def start_job(reference, job, &block)
-      new(reference, job).tap do |job_execution|
-        if enabled
-          registry[job.id] = job_execution
-          job_execution.start!(&block)
-          ActiveSupport::Notifications.instrument "job.threads", thread_count: registry.length
-        end
-      end
+    def active?(id, key: id)
+      registry.active?(key, id)
+    end
+
+    def queued?(id, key: id)
+      registry.queued?(key, id)
+    end
+
+    def start_job(reference, job, key: job.id, **options, &block)
+      registry.add(key, reference, job, options, &block)
     end
```

Comment:
Last comment, I think.  :-)

I find the method signature here confusing.  The `**options` double-splat has one special option (`:on_complete`), and all other values in the hash are env variables.  There's the `:on_complete` block that is run afterwards, and then another block that can be the main execution block of the job.  How about we just put everything in a hash, like:

```ruby
def start_job(reference, job, key: job.id, env: {}, on_exec: nil, on_complete: nil)
```